src/overtone/sc/sclang.clj

```diff
@@ -0,0 +1,422 @@
+(ns overtone.sc.sclang
+  (:require
+   [babashka.process :as proc]
+   [clojure.edn :as edn]
+   [clojure.java.io :as io]
+   [clojure.string :as str]
+   [overtone.helpers.system :refer [get-os linux-os? mac-os? windows-os?]]
+   [overtone.sc.machinery.server.connection :as ov.conn]
+   [overtone.sc.synth :as ov.synth]))
+
+(defonce *sclang-user-path (atom nil))
+
+(defn sclang-path
+  "The user must have SuperCollider installed.
+
+  For sclang location in each OS, see
+  https://github.com/supercollider/supercollider/wiki/Path-searching#scide-finds-sclang"
+  []
+  (or @*sclang-user-path
+      (cond
+        (mac-os?)
+        "/Applications/SuperCollider.app/Contents/MacOS/sclang"
+        ;; It's simply `sclang` in non OSX environments,
+        ;; https://github.com/supercollider/supercollider/wiki/Path-searching#scide-finds-sclang
+        ;; TODO Someone should test on Linux and Windows to confirm that this works.
+        (windows-os?)
+        "sclang.exe"
+
+        :else
+        "sclang")))
+
+(defn transpile
+  "Converts hiccup-like syntax to a SC string."
+  [sc-clj]
+  (let [[op & body] (if (sequential? sc-clj)
+                      sc-clj
+                      [sc-clj])]
+    (try
+      (cond
+        (and (= op :SynthDef) (sequential? sc-clj))
+        (let [{:keys [args resources-dir]
+               :or {resources-dir "resources"}
+               synthdef-name :name}
+              (first body)
+
+              synthdef-name (-> synthdef-name
+                                (str/replace #"\." "_")
+                                (str/replace #"/" "_"))
+
+              resource-prefix "sc/synthdef"
+              file-dir (str resources-dir "/" resource-prefix)
+
+              file-path (str file-dir "/" synthdef-name ".scsyndef")
+              resource-path (str resource-prefix "/" synthdef-name ".scsyndef")
+
+              metadata-file-path (str file-dir "/" synthdef-name ".edn")
+              metadata-resource-path (str resource-prefix "/" synthdef-name ".edn")
+
+              body (rest body)
+              synthdef-str (str "SynthDef"
+                                "(" (str "\"" synthdef-name "\"" ",") " {\n"
+                                (->> [(str "arg "
+                                           (->> args
+                                                (mapv (fn [[arg-identifier default]]
+                                                        (str (name arg-identifier) "=" default)))
```

Comment:
Should we transpile `default`?

Reply:
If we do we might want to wrap it in parens:
- "arg" style, default value is a literal but you want to treat it like an expression: { arg x = (2); ... } [[2]](http://doc.sccode.org/Reference/Functions.html#[2])

http://doc.sccode.org/Reference/Functions.html

Reply:
Never used a default that was not a number, bu it won't hurt, done it, thanks, man!!